zerver/management/commands/create_user.py

```diff
@@ -56,14 +58,27 @@ def handle(self, *args: Any, **options: Any) -> None:
         try:
             if options["password_file"] is not None:
                 with open(options["password_file"]) as f:
-                    pw = f.read().strip()
+                    pw: Optional[str] = f.read().strip()
             elif options["password"] is not None:
+                logging.warning(
+                    "Passing password on the command line is insecure; prefer --password-file."
+                )
                 pw = options["password"]
             else:
+                # initial_password will return a random password that
+                # is a salted hash of the email address in a
+                # development environment, and None in a production
+                # environment.
```

Comment:
Do we want an assertion also in the actual function:
```
def initial_password(email: str) -> Optional[str]:
    """Given an email address, returns the initial password for that account, as
    created by populate_db."""

    if settings.INITIAL_PASSWORD_SALT is not None:
        encoded_key = (settings.INITIAL_PASSWORD_SALT + email).encode()
        digest = hashlib.sha256(encoded_key).digest()
        return base64.b64encode(digest)[:16].decode()
    else:
        # None as a password for a user tells Django to set an unusable password
        return None
```
default_settings has:
```
# Random salt used when deterministically generating passwords in
# development.
INITIAL_PASSWORD_SALT: Optional[str] = None
```
so there's the assumption that this is None outside of DEVELOPMENT, but it feels like we should enforcing that assumption at the fundamental level to avoid misuse in weird circumstances?

Reply:
Yeah, probably an `assert not settings.PRODUCTION` is probably a good idea.

Reply:
Added a commit doing this.

Reply:
I might need `assert settings.DEVELOPMENT` or to remove this, because we mock `settings.PRODUCTION` in a few tests.